src/gr11review/part2/Utility.java

```diff
@@ -1,5 +1,621 @@
 package gr11review.part2;
+import java.util.*;
+import java.io.*;
+
 
 public class Utility {
+
+  /**
+  * This method takes a string input and if the letters z and p
+  * exsist in the string with only one letter between them, then 
+  * that letter gets taken out of the string.
+  *
+  * @param str  This is the string given
+  * @return this method returns a String that does not include 
+  * the letters between the z's and p's in the original string
+  * @author Kevin Basta
+  */
+  public static String zipZap(String str) {
+    String strLetterBefore;
+    String strFirstLetter;
+    String strSecondLetter;
+    String strReturnPattern;
+    strReturnPattern = "";
+    int i;
+    i = 0;
+
+    for (i = 0; i <= str.length() - 2; i++) {
+      strFirstLetter = str.substring(i, i + 1);
+      
+      if (i == 0) {
+        strReturnPattern += strFirstLetter;
+      } else {
+        strSecondLetter = str.substring(i + 1, i + 2);
+        strLetterBefore = str.substring(i - 1, i);
+        
+        if (strLetterBefore.equalsIgnoreCase("z") && strSecondLetter.equalsIgnoreCase("p")) {
+          
+        } else {
+          strReturnPattern += strFirstLetter;
+        }
+
+        if (i == str.length() - 2) {
+          strReturnPattern += strSecondLetter;
+        }
+      }
+    }
+
+    return strReturnPattern;
+  }
+  
+
+  /**
+  * This method is given a txt file an it determines what is the
+  * longest word in that file
+  *
+  * @param filenametxt  This is the name of the text file that contains the words
+  * @return this method returns the longes word it found in the text file
+  * @author Kevin Basta
+  */
+  public static String longestWord(String filenametxt) throws IOException{
+    BufferedReader textFile = new BufferedReader(new FileReader("src/gr11review/part2/" + filenametxt));
```

Comment:
@KevinB-school Don't assume the filepath, assume it comes through as part of the the `filenametext` parameter